### PR TITLE
Added recipe for `Opacus`

### PR DESCRIPTION
**Pull request by sarthakpati**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sarthakpati** [comment]
Hi @conda-forge/python, this is ready for review!

File: recipes/opacus/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/opacus-{{ version }}.tar.gz
+    sha256: c84fcbebe01bb185d3d2abcad135c47687afa93a0c7f6fd92d0018934b039cb9
+  - url: https://raw.githubusercontent.com/pytorch/opacus/main/dev_requirements.txt
```
Comment (jaimergp) [inline recipes/opacus/meta.yaml:12]: Why is this needed? 🤔

Comment (sarthakpati) [reply recipes/opacus/meta.yaml:12]: The setup required the `dev_requirements` file, for some reason. The recipe would not build without this.

Comment (sarthakpati) [reply recipes/opacus/meta.yaml]: Improved this to use `https://raw.githubusercontent.com/pytorch/opacus/v{{ version }}/dev_requirements.txt`.

Comment (dopplershift) [reply recipes/opacus/meta.yaml]: That can be fine temporarily, but the issue about it missing needs to be opened with upstream.

Comment (sarthakpati) [reply recipes/opacus/meta.yaml]: Agreed, I'm going to put a PR once this is merged.

**Review by dopplershift — APPROVED** [review]

**Comment by dopplershift** [comment]
Ignoring noarch Windows build failure due to missing dependency.
